Persistent Storage changelog (#2287)
FlowFuse/website: src/changelog/2024/07/persistent-storage.md

---

> title: Persistent Storage
> description:

Reviewer: ```suggestion
title: Persistent Storage on FlowFuse Cloud
description: Bring persistent storage to FlowFuse Cloud and self-hosted Kubernetes platforms
```

date: 2024-07-04 13:00:00.0
authors: ["ben-hardill"]
tags:
    - changelog
---

> Users of FlowFuse Cloud and self hosting users on Kubernetes can now make use of a 

Reviewer: ```suggestion
Users of FlowFuse Cloud and self-hosted users on Kubernetes can now make use of a 
```

new persistent storage solution.

Starting with the v2.6.0 release the Pods running the Node-RED Instances will have 
a Persistent Volume mounted on `/data/storage` in which files can be written. These
files will persist for the lifetime of the Instance including across Susspend/Resume
and Stack upgrades.

> This update means nodes like node-red-node-sqlite can be used to store data safely.

Reviewer: ```suggestion
This update means nodes like [node-red-node-sqlite](https://flows.nodered.org/node/node-red-node-sqlite) can be used to store data safely.
```

> This capability replaces the current FlowFuse FileServer which makes use of custom 
> versions of the Node-RED code File nodes to store and read files from a remote network
> store.

Reviewer: ```suggestion
This capability replaces the current FlowFuse FileServer which made use of custom 
versions of the Node-RED code File nodes to store and read files from a remote network
store. However this didn't allow 3rd party nodes to also benefit from the storage provided.
```

> New instances created will default to use the new Persistent Storage volumes, users 
> with exising instances making use of the existing FileServer file storage will be able
> to switch over by upgrading to the latest version of the Node-RED Stack and then contacting
> Support who will be able to arrange for existing files to be moved between to the new 
> volume.

Reviewer: ```suggestion
New instances created will default to use the new Persistent Storage volumes. If you want to upgrade an existing instance that uses the FileServer storage, get in touch with [Support](https://flowfuse.com/contact-us/) and we can help migrate your data over.
 ```
